src/routes/Groups/Groups.js

```diff
@@ -1,125 +1,110 @@
-import React, { Component } from 'react'
 import PropTypes from 'prop-types'
+import { Link } from 'react-router-dom'
+import React, { Component } from 'react'
 import { DEFAULT_BANNER, DEFAULT_AVATAR } from 'store/models/Group'
-import Dropdown from 'components/Dropdown'
 import { bgImageStyle } from 'util/index'
-import Icon from 'components/Icon'
-import TextInput from 'components/TextInput'
+import Button from 'components/Button'
 import RoundImage from 'components/RoundImage'
-import { find } from 'lodash/fp'
-import { Link } from 'react-router-dom'
 import { groupUrl } from 'util/navigation'
 
 import './Groups.scss'
 
-// import ScrollListener from 'components/ScrollListener'
-// import { CENTER_COLUMN_ID } from 'util/scrolling'
-
-const { array } = PropTypes
-
-const sortOptions = [
-  { id: 'name', label: 'Alphabetical' },
-  { id: 'num_members', label: 'Popular' },
-  { id: 'created_at', label: 'Newest' }
-]
-
 export default class Groups extends Component {
```

Comment:
This would make a tidy functional component with a `useState()` for the invite picker... In lock-step with React moves I try and make new components that can be (which is most) functional components using hooks. It hurts a little at first, but starts to feel good soon after. Put here for discussion, not thinking you'd change this component here now.

Reply:
i know, i too want to move more towards functional components, there's just some momentum i have towards what i know a bit better